### ISSUE TITLE
[iOS Protection Dashboard] [Accessibility] - Close button is not announced with voice over

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#35714]
**Steps to Reproduce**

1. Turn voice over On.
2. Launch Firefox and visit a website.
3. Go to homepage and tap on tracker blocker pill.
4. Go through all the elements of the dashboard.

**Expected Behavior**

“x” button is  announced with voice over.

**Actual Behavior**

“x” button is not announced with voice over.

**Device & Build Information**

- Device: iPhone 15 (17.6)
- Build version: v9000(76985)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-16841)
